client/galaxy/scripts/mvc/history/hdca-li.js

```diff
@@ -43,13 +47,98 @@ var HDCAListItemView = _super.extend(
         _swapNewRender: function($newRender) {
             _super.prototype._swapNewRender.call(this, $newRender);
             //TODO: model currently has no state
-            var state = !this.model.get("populated") ? STATES.RUNNING : STATES.OK;
-            //if( this.model.has( 'state' ) ){
+            var state;
+            var jobStatesSummary = this.model.jobStatesSummary;
+            if (jobStatesSummary) {
+                if (jobStatesSummary.new()) {
+                    state = "new";
+                } else if (jobStatesSummary.errored()) {
+                    state = "error";
+                } else if (jobStatesSummary.terminal()) {
+                    state = "ok";
+                } else if (jobStatesSummary.running()) {
+                    state = "running";
+                } else {
+                    state = "queued";
+                }
+            } else if (this.model.get("job_source_id")) {
+                // Initial rendering - polling will fill in more details in a bit.
+                state = STATES.NEW;
+            } else {
+                state = this.model.get("populated_state") ? STATES.OK : STATES.RUNNING;
+            }
             this.$el.addClass(`state-${state}`);
-            //}
+            var stateDescription = this.stateDescription();
+            console.log(stateDescription);
+            this.$(".state-description").html(stateDescription);
             return this.$el;
         },
 
+        stateDescription: function() {
+            var collection = this.model;
+            var elementCount = collection.get("element_count");
+            console.log(elementCount);
+            var jobStateSource = collection.get("job_source_type");
+            var collectionType = this.model.get("collection_type");
+            var collectionTypeDescription;
+            if (collectionType == "list") {
+                collectionTypeDescription = "list";
+            } else if (collectionType == "paired") {
+                collectionTypeDescription = "dataset pair";
+            } else if (collectionType == "list:paired") {
+                collectionTypeDescription = "list of pairs";
+            } else {
+                collectionTypeDescription = "nested list";
+            }
+            var itemsDescription = "";
+            if (elementCount == 1) {
+                itemsDescription = ` with 1 item`;
+            } else if (elementCount) {
+                itemsDescription = ` with ${elementCount} items`;
+            }
+            var collectionCount = 42; // TODO - and handle unpopulated...
```

Comment:
the default value seems confusing here

Reply:
Yes - I need to fix this - debugging left overs.